rules/no-new-buffer.js

```diff
@@ -1,40 +1,91 @@
 'use strict';
+const {getStaticValue} = require('eslint-utils');
 const getDocumentationUrl = require('./utils/get-documentation-url');
+const isNewExpressionWithParentheses = require('./utils/is-new-expression-with-parentheses');
 
-const MESSAGE_ID = 'no-new-buffer';
+const ERROR = 'error';
+const ERROR_UNKNOWN = 'error-unknown';
+const SUGGESTION = 'suggestion';
 const messages = {
-	[MESSAGE_ID]: '`new Buffer()` is deprecated, use `Buffer.{{method}}()` instead.'
+	[ERROR]: '`new Buffer()` is deprecated, use `Buffer.{{method}}()` instead.',
+	[ERROR_UNKNOWN]: '`new Buffer()` is deprecated, use `Buffer.alloc()` or `Buffer.from()` instead.',
+	[SUGGESTION]: 'Switch to `Buffer.{{method}}()`.'
 };
 
-const inferMethod = arguments_ => {
-	if (arguments_.length > 0) {
-		const [firstArgument] = arguments_;
+const inferMethod = (bufferArguments, scope) => {
+	if (bufferArguments.length !== 1) {
+		return 'from';
+	}
+
+	const [firstArgument] = bufferArguments;
+	if (firstArgument.type === 'SpreadElement') {
+		return;
+	}
+
+	if (firstArgument.type === 'ArrayExpression' || firstArgument.type === 'TemplateLiteral') {
+		return 'from';
+	}
+
+	const staticResult = getStaticValue(firstArgument, scope);
+	if (staticResult) {
+		const {value} = staticResult;
+		if (typeof value === 'number') {
+			return 'alloc';
+		}
+
 		if (
-			firstArgument.type === 'Literal' &&
-			typeof firstArgument.value === 'number'
+			typeof value === 'string' ||
+			Array.isArray(value)
 		) {
-			return 'alloc';
+			return 'from';
 		}
 	}
-
-	return 'from';
 };
 
+function fix(node, sourceCode, method) {
+	return function * (fixer) {
+		const [start] = node.range;
+		let end = start + 3; // `3` = length of `new`
+		const textAfter = sourceCode.text.slice(end);
+		const [leadingSpaces] = textAfter.match(/^\s*/);
+		end += leadingSpaces.length;
```

Comment:
This is the exact same as https://github.com/sindresorhus/eslint-plugin-unicorn/blob/ca2f54f609ee46bf68503accfddb5b4ca31755d8/rules/new-for-builtins.js#L55-L59 Maybe it could be reused?

Reply:
I'll do a clean up later, because I don't want add a function to do `removeNewAndSpaces`, I want add `removeLeadingSpaceAfter(index)`, I think there are other places can use it.

Reply:
I searched codebase, seems only these two places, I'll do it now.